python/src/deltachat/provider.py

```diff
@@ -0,0 +1,41 @@
+""" Provider info class """
+
+import deltachat
+from .capi import ffi, lib
+from .cutil import as_dc_charpointer, from_dc_charpointer, iter_array
+
+
+class Provider(object):
+    def __init__(self, email):
+        self._provider = lib.dc_get_provider_info_from_email(as_dc_charpointer(email))
+
+    @classmethod
+    def from_email(cls, email):
+        instance = cls(email)
+        # print(instance._provider)
+        if instance._provider == ffi.NULL:
+            return None
+        else:
+            return instance
```

Comment:
... then thiscould beocme 
```
provider = lib.dc_get_provider_info_from_email(as_dc_charpointer(email))
if provider == ffi.NULL: 
     return None
return Provider(provider)
```

Otherwise you have two constructors (__init__ and from_email) that take the same argument which is confusing.